spikeinterface/extractors/nwbextractors.py

```diff
@@ -54,118 +72,131 @@ class NwbRecordingExtractor(BaseRecording):
     mode = 'file'
     installation_mesg = "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
 
-    def __init__(self, file_path: PathType, electrical_series_name: str = None, load_time_vector=False):
-        """
-        Load an NWBFile as a RecordingExtractor.
-
-        Parameters
-        ----------
-        file_path: path to NWB file
-        electrical_series_name: str, optional
-        """
-        assert HAVE_NWB, self.installation_mesg
+    def __init__(self, file_path: PathType, electrical_series_name: str = None, load_time_vector: bool = False,
+                 samples_for_rate_estimation: int = 100000):
+        check_nwb_install()
         self._file_path = str(file_path)
         self._electrical_series_name = electrical_series_name
-        with NWBHDF5IO(self._file_path, mode='r',load_namespaces=True) as io:
-            nwbfile = io.read()
-            es = get_electrical_series(nwbfile,self._electrical_series_name)
-            if hasattr(es, 'timestamps') and es.timestamps:
-                sampling_frequency = 1. / np.median(np.diff(es.timestamps))
-                t_start = es.timestamps[0]
-                if load_time_vector:
-                    times_kwargs = dict(time_vector=es.timestamps)
-                else:
-                    times_kwargs = dict(sampling_frequency=sampling_frequency, t_start=t_start)
-            else:
-                sampling_frequency = es.rate
-                if hasattr(es, 'starting_time'):
-                    t_start = es.starting_time
+        
+        io =  NWBHDF5IO(self._file_path, mode='r',load_namespaces=True)
+        self._nwbfile = io.read()
+        self._es = get_electrical_series(
+            self._nwbfile, self._electrical_series_name)
+        
+        sampling_frequency = None
+        if hasattr(self._es, 'rate'):
+            sampling_frequency = self._es.rate
+        
+        if hasattr(self._es, 'starting_time'):
+            t_start = self._es.starting_time
+        else:
+            t_start = 0.
+        
+        timestamps = None
+        if hasattr(self._es, 'timestamps'):
```

Comment:
and load_time_vector

Reply:
no this is ok because only the object reference is accessed (and the correct `t_start`)